ocw/lib/gce.py

```diff
@@ -91,13 +91,17 @@ def list_regions(self) -> list:
         return [region["name"] for region in regions]
 
     def list_zones(self, region) -> list:
-        region = (
-            self.compute_client()
-            .regions()
-            .get(project=self.project, region=region)
-            .execute()
-        )
-        return [basename(z) for z in region["zones"]]
+        try:
+            region = (
```

Comment:
Why not getting the regions and check, if the desired region is in the list? This error message looks like spam to me.

Reply:
> Why not getting the regions and check, if the desired region is in the list? This error message looks like spam to me.

This is dynamic as regions and zones can suddenly "disappear".  That's why they can't be cached.  Also, the code that calls `list_zones()` iterates over `list_regions()`.  We wouldn't want to call `list_regions()` twice.

Reply:
ok